tools/proto-js-plugin/src/main/java/io/spine/js/generate/CodeLines.java

```diff
@@ -266,49 +270,47 @@ public void ifNotNullOrUndefined(String value) {
      */
     public void exportMap(String mapName) {
         checkNotNull(mapName);
-        addLine("module.exports." + mapName + " = new Map([");
-        currentDepth++;
-    }
-
-    /**
-     * Adds an entry to the map.
-     *
-     * <p>Assumes that the cursor is currently inside the exported map declaration.
-     *
-     * @param entry
-     *         the entry to add
-     * @param isLast
-     *         whether this entry is last or there will be more
-     */
-    public void addMapEntry(String entry, boolean isLast) {
-        checkNotNull(entry);
-        if (isLast) {
-            addLine(entry);
-        } else {
-            addLine(entry + ',');
-        }
+        append("module.exports." + mapName + " = new Map([");
+        increaseDepth();
     }
 
     /**
      * Exits the exported map declaration.
      */
     public void quitMapDeclaration() {
         decreaseDepth();
-        addLine("]);");
+        append("]);");
     }
 
     /**
      * Manually increases the current depth.
      */
     public void increaseDepth() {
-        currentDepth++;
+        currentDepth = currentDepth.incremented();
     }
 
     /**
      * Manually decreases the current depth.
      */
     public void decreaseDepth() {
-        currentDepth--;
+        currentDepth = currentDepth.decremented();
+    }
+
+    /**
+     * Merges lines by addition of a comma to each line except the last one.
+     */
+    public static CodeLines commaSeparated(List<CodeLine> lines) {
```

Comment:
Please use a `Joiner` instead.

Reply:
As discussed, a `Joiner` doesn't suite here since the method works with lines, which are appended to `CodeLines`.